addons/gloot/ctrl_inventory_grid.gd

```diff
@@ -314,18 +334,62 @@ func _select(item: InventoryItem) -> void:
     _selected_item = item
     _refresh_selection()
 
+func switch_virtual_cursor_grid(new_grid):
+    if not _grabbed_with_virtual_cursor or not new_grid:
+        return
+
+    _virtual_cursor_gird = new_grid
+    _virtual_cursor = new_grid.rect_global_position
+
 
 func _input(event: InputEvent) -> void:
-    if !(event is InputEventMouseButton):
+    if event.is_action_released("ui_cancel"):
+        _cancel_grab()
         return
 
-    if !_grabbed_ctrl_inventory_item:
+    _handle_virtual_cursor(event)
+    _handle_mouse(event)
+
+func _handle_virtual_cursor(event: InputEvent):
+    if not _grabbed_with_virtual_cursor:
+        return
+
+    var delta
```

Comment:
I think we should be able to declare `delta` as a `Vector2` if we reorganize this function a little bit: Move the `ui_accept` check to the top and use `elif` when checking for directional keys. That way we can return from the function if none of them was pressed and there's no need to check if `delta` is `null`.